tests/drf_generator_test.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""Tests for drf_generator package."""
+
+from pathlib import Path
+
+import pytest
+
+import drf_generator.generator as drf_generator
+
+parent_path = Path(__file__).parent
+first_validation_file = parent_path.joinpath('test_validate1.txt').resolve()
+second_validation_file = parent_path.joinpath(
+    'test_validate2.txt').resolve()
+
+
+def equal(first, second):
+    """
+    Check the equivalence of two objects.
+
+    :param first: First object to compare.
+    :param second: Second object to compare.
+    :return: True if the objects are equal, False otherwise.
+    """
+    return first == second
+
+
+@pytest.mark.parametrize('first,second,expected,comparator', [
+    ('A', 'F', ['A', 'B', 'C', 'D', 'E', 'F'], equal),
+    ('F', 'A', [], equal),
+    ('1', '6', ['1', '2', '3', '4', '5', '6'], equal),
+    ('6', '1', [], equal),
+])
+def test_char_range(first, second, expected, comparator):
+    """
+    Test the char_range function.
+
+    :param first: First character.
+    :param second: Second character.
+    :param expected: Expected result.
+    :param comparator: Comparator function.
+    """
+    result = list(drf_generator.char_range(first, second))
+    assert comparator(result, expected)
+
```

Comment:
Why isn't just
```python
assert result == expected
```
I see you'd want to leave room for testing more than one comparator, but it just makes harder to figure out what the test is actually testing, and in failure case, it's a bit convoluted to see the error.
If we want to make another type of list comparison (like inclusion, or permutation), it's ok to write another test case, or even a conditional. IMO it's an unneeded abstraction.